content/en/blog/2020/mosn-proxy/index.md

```diff
@@ -0,0 +1,115 @@
+
+## What is the difference between MOSN and Istio default proxy? What are the advantages of MOSN?
+
+### Differences in language stacks
+
+MOSN is written in GoLang. GoLang has strong guarantees in terms of production efficiency and memory security. At the same time, GoLang has an extensive library ecosystem in the cloud-native era. The performance is acceptable and usable in the service mesh scenario. Therefore, MOSN has a lower intellectual cost for companies and individuals using languages such as GoLang and Java.
+
+### Differentiation of core competence
+
+- MOSN supports a multi-protocol framework, and users can easily access private protocols with a unified routing framework.
+- Multi-process plug-in mechanism, which can easily extend the plug-ins of independent MOSN processes through the plug-in framework, and do some other management, bypass and other functional module extensions.
+- Transport layer national secret algorithm support with Chinese encryption compliance, etc.
+
+## MOSN with Istio
+
+MOSN can be used not only as a stand-alone Layer 4/Layer 7 load balancer, but can also be integrated into Istio as a sidecar proxy or ingress gateway in Kubernetes. The following is an introduction to the use of MOSN as an Istio data plane.
+
+## Setup Istio
+
+You can download a zip file for your operating system from the [Istio release](https://github.com/istio/istio/releases/tag/1.5.2) page. This file contains: the installation file, examples and the `istioctl` command line tool.
+To download Istio (this example uses Istio 1.5.2) uses the following command.
+
+{{< text bash >}}
+$ export ISTIO_VERSION=1.5.2
+$ curl -L {{< github_file >}}/release/downloadIstioCandidate.sh | sh -
+{{< /text >}}
+
+The downloaded Istio package is named `istio-1.5.2` and contains:
+- `install/kubernetes`: Contains YAML installation files related to Kubernetes.
+- `examples/`: Contains example applications.
+- `bin/`: Contains the istioctl client files.
+
+Switch to the folder where of Istio located.
+
+{{< text bash >}}
+$ cd istio-$ISTIO_VERSION/
+{{< /text >}}
+
+Add the `istioctl` client path to `$PATH` with the following command.
+
+{{< text bash >}}
+$ export PATH=$PATH:$(pwd)/bin
+{{< /text >}}
+
+Until now, it has been possible to flexibly customize the Istio control plane and data plane configuration parameters via the `istioctl` command line tool.
+
+## Setting MOSN as the Data Plane
+
+MOSN is specified as the data plane in Istio via the parameters of the istioctl command.
+
+{{< text bash >}}
+$ istioctl manifest apply  --set .values.global.proxy.image="mosnio/proxyv2:1.5.2-mosn"   --set meshConfig.defaultConfig.binaryPath="/usr/local/bin/mosn"
+{{< /text >}}
+
+Check that Istio-related pod services are deployed successfully.
+
+{{< text bash >}}
+$ kubectl get svc -n istio-system
+{{< /text >}}
+
+If the service `STATUS` is Running, then Istio has been successfully installed and you can deploy the Bookinfo sample later.
+
+## Bookinfo Examples
```

Comment:
The post tails off here.  Your message is "At this point, you can do everything you would do with Envoy", but aside from proving it works, could they do something else here?

Is there a difference in how you debug?  Could you inspect a container when Bookinfo is deployed?  Could you talk about how to get logs, etc?

Reply:
> The post tails off here. Your message is "At this point, you can do everything you would do with Envoy", but aside from proving it works, could they do something else here?
> 
> Is there a difference in how you debug? Could you inspect a container when Bookinfo is deployed? Could you talk about how to get logs, etc?

Done.